api/schema/gcp_flow_logs_pub_sub_resource.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Illumio, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+var (
+	gcpFlowLogsPubSubResource = Resource{
+		TypeName: "gcp_flow_logs_pub_sub",
+		Schema: resource_schema.Schema{
+			Version:     1,
+			Description: "Manages CloudSecure access to flow logs in a GCP Pub/Sub.",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"project_id": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						MarkdownDescription: "GCP project ID.",
+						Required:            true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: ImmutableAttributeMode,
+					},
+				},
+				"pub_sub_topic_resource_id": StringResourceAttributeWithMode{
```

Comment:
1. `pub_sub` -> `pubsub` to be consistent with the `google` Terraform provider.
2. The topic's ID attribute is spelled simply `id` in the `google` Terraform provider schema, not `resource_id`. Be consistent. Cf. https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/pubsub_topic#id-1
```suggestion
				"pubsub_topic_id": StringResourceAttributeWithMode{
```